core/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -175,6 +206,16 @@ public String concreteSingleIndex(ClusterState state, IndicesRequest request) {
     }
 
     /**
+     * @return whether the specified alias or index exists. If the alias or index contains datemath then that is resolved too.
+     */
+    public boolean hasIndexOrAliasAndResolveDateMath(String aliasOrIndex, ClusterState state) {
+        assert expressionResolvers.get(0) instanceof DateMathExpressionResolver;
+        List<String> resolvedDateMath = expressionResolvers.get(0).resolve(new Context(state, IndicesOptions.lenientExpandOpen()), Collections.singletonList(aliasOrIndex));
+        assert resolvedDateMath.size() == 1;
```

Comment:
I am not sure we should have this assert here. this method should only say whether an index is there or not. If it's there we are good, otherwise we go and auto create it. If the expression resolves to multiple indices is something that we have to deal with later on, based on the indices options. Makes sense?